Parallel tests

blueyed opened the pull request:
1. cmake: tests: do not use USES_TERMINAL

This is an intermediate/first step.
It will currently just buffer all output, but this is required when
having multiple RunTests instances in parallel.

It also fixes "sh" hanging for ~40s with `abort()` in `children_kill_cb`
with:

> make functionaltest TEST_FILTER='trapping SIGTERM' TEST_FILE=test/functional/autocmd/termclose_spec.lua

Ref: https://github.com/neovim/neovim/pull/4753#issuecomment-503221871

blueyed committed:
cmake: tests: do not use USES_TERMINAL

This is an intermediate/first step.
It will currently just buffer all output, but this is required when
having multiple RunTests instances in parallel.

It also fixes "sh" hanging for ~40s with `abort()` in `children_kill_cb`
with:

> make functionaltest TEST_FILTER='trapping SIGTERM' TEST_FILE=test/functional/autocmd/termclose_spec.lua

Ref: https://github.com/neovim/neovim/pull/4753#issuecomment-503221871

justinmk commented:
> also fixes "sh" hanging for ~40s with abort() in children_kill_cb
with:

But why does it abort?

blueyed commented:
> But why does it abort?

Because that's sometimes my C was of doing `pdb.set_trace()`.. ;)  (i.e. I've added it there to see if this code is reached, and being able to inspect it using `coredumpctl gdb` then)

blueyed committed:
RunTests.cmake: use scratch dir per invocation

The plan is to pass subdirectories for different test types via
TEST_PATH directly (it appears to be overkill to split this by actual
test files).
